constants/globals.js

```diff
@@ -1,9 +1,11 @@
 "use strict";
 
 const CONFIG_FILE   = "cleave-html-config.json";
-const FILE_TYPE     = "utf8"; 
+const FILE_TYPE     = "utf8";
+const EOF           = "EOF";
```

Comment:
Why aren't you using ASCII EOF? That way it's one character

Reply:
ascii doesn't have a EOF "character". It is represented by `-1`. We can set EOF to `-1` if you want.

Reply:
not sure if -1 is the right solution

But you could call `isValidTagCharacter('EOF')` and it would return true even though it's not a character, it's a string. Same inconsistency with `nextChar()`

Reply:
Neither of those will receive or output more than a character at a time, but I understand what you're saying. An empty string also won't work because nextChar can return and empty string and should in some cases. `-1` might be a good solution since it is a completely different type from anything that is given as input or received as output. I can change it to `-1` for the time being until we arrive at a better solution. Does that work with you?

Reply:
I think false would probably be better so you could do `if (peak()) {}` 

I believe `nextChar()` could return `EOF` which is a string right?

Reply:
`while(hasNext()) {}` would be a nice way to use it, so I second the use of false

Reply:
@drewatk Yes, I mispoke it does return EOF. 
Sounds good. I'll set it to false.

Reply:
@vritant24 I would advise against using `while(peek()) {}` since this would terminate if peek() returned `" "`. The current peek implementation only returns non space characters or EOF. So, peek may be safe, but peekChar would not be.

Reply:
That makes sense.